privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -654,6 +654,60 @@ def testconnection(cls, param):
         
         return success, desc
 
+    def add_user(self, uid, object_class=None, attributes=None):
+        """
+        Add a new user to the LDAP directory.
+
+        attributes are these
+        "username", "surname", "givenname", "email",
+        "mobile", "phone", "password"
+
+        :param uid: The uid of the user object in the resolver
+        :type uid: string
+        :param object_class: Attributes according to the attribute mapping
+        :type object_class: list
+        :param attributes: Attributes according to the attribute mapping
+        :type attributes: dict
+        :return: The new UID of the user. The UserIdResolver needs to
+        determine the way how to create the UID.
+        """
+        attributes = attributes or {}
+        try:
+            self._bind()
+
+            params = self._attributes_to_ldap_attributes(attributes)
+            self.l.add(uid, object_class, params)
```

Comment:
You need to call

    self.l.add(self._getDN(uid), object_class, params)

Otherwise it will only work, if the uidtype is DN. If you are using uidtype=objectGUID, the uid will be the objectGUID and it will fail.